core/src/main/scala/org/http4s/headers/Origin.scala

```diff
@@ -50,31 +49,36 @@ object Origin {
       toUri.render(writer)
   }
 
-  private[http4s] val parser: Parser0[Origin] = {
-    import Parser.{`end`, char, string, until}
-    import Rfc5234.{alpha, digit}
-
-    val unknownScheme =
-      alpha ~ Parser.oneOf(List(alpha, digit, char('+'), char('-'), char('.'))).rep0
-    val http = string("http")
-    val https = string("https")
-    val scheme = List(https, http, unknownScheme)
-      .reduceLeft(_ orElse _)
-      .string
-      .map(Uri.Scheme.unsafeFromString)
-    val stringHost = until(char(':').orElse(`end`)).map(RegName.apply)
-    val bracketedIpv6 = char('[') *> Uri.Parser.ipv6Address <* char(']')
-    val host = List(bracketedIpv6, Uri.Parser.ipv4Address, stringHost).reduceLeft(_ orElse _)
+  private[http4s] val nullHostParser: Parser0[Origin.Null.type] = {
+    import Parser.{`end`, string}
+
+    (string("null") *> `end`).as(Origin.Null)
+  }
+
+  private[http4s] val singleHostParser: Parser[Origin.Host] = {
+    import Parser.{char, string}
+    import Rfc5234.digit
+
+    val scheme = Uri.Parser.scheme
+
+    val host = Uri.Parser.host
+
     val port = char(':') *> digit.rep.string.map(_.toInt)
-    val nullHost = (string("null") *> `end`).as(Origin.Null)
 
-    val singleHost = ((scheme <* string("://")) ~ host ~ port.?).map { case ((sch, host), port) =>
+    ((scheme <* string("://")) ~ host ~ port.? ~ char('/').?).map { case (((sch, host), port), _) =>
```

Comment:
Is that `/` actually allowed by the specification? I had a look at https://www.rfc-editor.org/rfc/rfc6454#section-7 and I don't see it there.